### PR TITLE
Picking up work in PR  #205

### PR DESCRIPTION
**Pull request by jhughes24816**
Motivation:

Add ACL functionality to vertx-kafka-client

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

**Comment by jhughes24816** [comment]
@vietj @ppatierno Can I get a review on this please?

File: src/main/java/io/vertx/kafka/admin/AclBinding.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.kafka.admin;
+
+import java.util.Objects;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
```
Comment (ppatierno) [inline src/main/java/io/vertx/kafka/admin/AclBinding.java:21]: blank line

**Review by ppatierno — COMMENTED** [review]
I know that it was already this way for others operations but as I asked in https://github.com/vert-x3/vertx-kafka-client/pull/205#issuecomment-890813935 why we cannot use the Kafka classes instead of creating new ones and then using the `Helper` to convert? @vietj said that it should be possible. At this point my only doubt is about changing the way or to stick with "duplicating" code.

File: src/main/java/io/vertx/kafka/admin/AclBinding.java (hunk 1 above)
Comment (ppatierno) [reply src/main/java/io/vertx/kafka/admin/AclBinding.java:21]: @vietj as the Vert.x expert, what's your opinion on this? how the other libraries are behaving?

Comment (jhughes24816) [reply src/main/java/io/vertx/kafka/admin/AclBinding.java:21]: @vietj  Can we get your input on this to clarify what is desired in this project? Thanks!

**Comment by jhughes24816** [comment]
@vietj @ppatierno Can this get a review please?

**Comment by jhughes24816** [comment]
@vietj  @ppatierno  I have resolved the merge conflicts introduced from the other PR please review :)

**Comment by vietj** [comment]
thanks, can you review this @ppatierno ?

**Comment by jhughes24816** [comment]
@vietj @ppatierno It would be good to close this out soon. Let me know if you need anything from me, happy to fix/adjust/refactors as required. Thanks!

File: src/main/java/io/vertx/kafka/admin/AclBinding.java (hunk 1 above)
Comment (vietj) [reply src/main/java/io/vertx/kafka/admin/AclBinding.java:21]: you should look at `@GenIgnore(GenIgnore.PERMITTED_TYPES)` which allow to use non vertx types

**Comment by vietj** [comment]
Can you investigate `@GenIgnore(GenIgnore.PERMITTED_TYPES)` to use kafka types ?

**Comment by jhughes24816** [comment]
I have pushed a commit that replaces the vertx copies of kafka objects with imports of the kafka objects. I have not added this annotation to anything since it was on all the acl methods in the interface already. Let me know if I have missed anything.

**Comment by jhughes24816** [comment]
@vietj @ppatierno Is there anything else left to be done here?

**Review by ppatierno — APPROVED** [review]
It's ok on my side thanks.

**Comment by vietj** [comment]
thanks for your contribution @jhughes24816

**Pull request merged**